src/wasm.h

```diff
@@ -503,6 +504,32 @@ class Literal {
       default: WASM_UNREACHABLE();
     }
   }
+  template <typename T>
+  static T rol(T val, T count) {
+    T mask = sizeof(T) * CHAR_BIT - 1;
+    count &= mask;
+    return (val << count) | (val >> (-count & mask));
+  }
+  Literal rotL(const Literal& other) const {
+    switch (type) {
+      case WasmType::i32: return Literal(rol(uint32_t(i32), uint32_t(other.i32)));
+      case WasmType::i64: return Literal(rol(uint64_t(i64), uint64_t(other.i64)));
+      default: WASM_UNREACHABLE();
+    }
+  }
+  template <typename T>
+  static T ror (T val, T count) {
+    T mask = sizeof(T) * CHAR_BIT - 1;
+    count &= mask;
+    return (val >> count) | (val << (-count & mask));
+  }
+  Literal rotR(const Literal& other) const {
+    switch (type) {
+      case WasmType::i32: return Literal(ror(uint32_t(i32), uint32_t(other.i32)));
+      case WasmType::i64: return Literal(ror(uint64_t(i64), uint64_t(other.i64)));
```

Comment:
everything looks good except I'm puzzled at these `ror` and `rol` methods - where are those provided?